Utils/azuretools-core/src/com/microsoft/azuretools/telemetry/AppInsightsClient.java

```diff
@@ -175,4 +212,62 @@ public static void createFTPEvent(String eventName, String uri, String appName,
     private static boolean isAppInsightsClientAvailable() {
         return configuration != null;
     }
+
+
+    public static void sendOpEnd(EventType eventType, String operName, Map<String, String> properties) {
+        sendOpEnd2(eventType, operName, properties, null);
+    }
+
+    public static void sendOpEnd(EventType eventType, String operName, Map<String, String> properties,
+        Map<String, Double> metrics) {
+        sendOpEnd2(eventType, operName, properties, metrics);
+    }
+
+    private static void sendOpEnd2(EventType eventType, String operName, Map<String, String> properties,
+        Map<String, Double> metrics) {
+        properties.put(OPERATION_NAME, operName);
+        properties.put(OPERATION_ID, UUID.randomUUID().toString());
+
+        String eventName = getEventName(eventType, EventName.opEnd);
+        create(eventName, null, properties, metrics, false);
+    }
+
+    public static void sendOpStart(EventType eventType, String operName, Map<String, String> properties) {
+        properties.put(OPERATION_NAME, operName);
+        properties.put(OPERATION_ID, UUID.randomUUID().toString());
+
+        String eventName = getEventName(eventType, EventName.opStart);
+        create(eventName, null, properties, false);
+    }
+
+    public static void sendError(EventType eventType, String operName, ErrorType errorType, String errMsg,
+        Map<String, String> properties) {
+        sendError2(eventType, operName, errorType, errMsg, properties, null);
+    }
+
+    public static void sendError(EventType eventType, String operName, ErrorType errorType, String errMsg,
+        Map<String, String> properties, Map<String, Double> metrics) {
+        sendError2(eventType, operName, errorType, errMsg, properties, metrics);
+    }
+
+    private static void sendError2(EventType eventType, String operName, ErrorType errorType, String errMsg,
```

Comment:
bad name

Reply:
Hard to give a nice name, I will give it a new name